stdlib/stdlib.mli

```diff
@@ -1369,6 +1369,12 @@ val at_exit : (unit -> unit) -> unit
    The functions are called in 'last in, first out' order: the
    function most recently added with [at_exit] is called first. *)
 
+(** {1 Type equality witness} *)
+
+type (_, _) eq = Refl: ('a, 'a) eq
```

Comment:
For the record: I'm not a fan of the `Refl` name, which is jargon, just like "lambda" is jargon.